File: angular_frontend/Project-three/src/app/model/hotelBooking.ts
export interface HotelBooking {
  hotelBookingId?: number | null;
  customerId: number;
  hotelId: number;
  checkInDate: string;
  checkOutDate: string;
  numRooms: number;
}

export interface HotelBookingRevicedDto {
  customerName: string;
  customerEmail: string;
  checkInDate: string;
  checkOutDate: string;
  numRooms: number;
}

File: travel_backed/src/main/java/com/example/travel/travel/entity/HotelBooking.java
package com.example.travel.travel.entity;

import java.util.Date;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

@Data
@RequiredArgsConstructor
@Entity
@Table(name = "HEMEL_HOTEL_BOOKING")
public class HotelBooking {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "customer_id")
    private Customer customer;

    @ManyToOne
    @JoinColumn(name = "hotel_id")
    private Hotel hotel;

    @ManyToOne
    @JoinColumn(name = "booking_id")  // ✅ This line links HotelBooking to Booking
    private Booking booking;

    private Date checkInDate;
    private Date checkOutDate;
    private Integer numRooms;
}
